pkg/deploy/prune/data.go

```diff
@@ -0,0 +1,155 @@
+package prune
+
+import (
+	"fmt"
+	"time"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+// DeploymentByDeploymentConfigIndexFunc indexes Deployment items by their associated DeploymentConfig, if none, index with key "orphan"
+func DeploymentByDeploymentConfigIndexFunc(obj interface{}) (string, error) {
+	controller, ok := obj.(*kapi.ReplicationController)
+	if !ok {
+		return "", fmt.Errorf("not a replication controller: %v", obj)
+	}
+	name, exists := controller.Annotations[deployapi.DeploymentConfigAnnotation]
```

Comment:
Change to --> deployutil.DeploymentConfigNameFor(controller)

Refer https://github.com/openshift/origin/pull/2227